### PR TITLE
A disabled build option should provide a proper description

### PR DESCRIPTION
**Pull request by bmuschko**
For more information see https://github.com/gradle/gradle/issues/3232

The code was there but must have gotten broken in some refactoring step. The disabled build option description wasn't properly tested.

### CODE REVIEW

**Comment by bmuschko** [comment]
@blindpirate Is this something you are planning to review or should I find someone else?

**Comment by blindpirate** [comment]
I'll finish it before your Monday, very sorry for the delay.

**Comment by bmuschko** [comment]
No worries, wasn't really on the plan for this sprint. It's definitely OK for me to find someone else to review if you are overloaded.

**Comment by bmuschko** [comment]
Still wondering if we could find a better way to represent a description for disabled boolean options. Right now it looks like this:

    --scan                    Creates a build scan. Gradle will emit a warning if the build scan plugin has not been applied. (https://gradle.com/build-scans) [incubating]
    --no-scan                 Disables option --scan. [incubating]

I think it would be better if the disabled option could have a more descriptive text. I might look into that again.

**Comment by bmuschko** [comment]
@blindpirate I improved the error messaging for disabled boolean options by being able to pass in a customized description. Can you please have another look?

**Comment by blindpirate** [comment]
LGTM.

File: subprojects/build-option/src/main/java/org/gradle/internal/buildoption/BooleanBuildOption.java [hunk 1]
```diff
@@ -49,17 +49,18 @@ public void applyFromProperty(Map<String, String> properties, T settings) {
 
     @Override
     public void configure(CommandLineParser parser) {
-        for (CommandLineOptionConfiguration config : commandLineOptionConfigurations) {
+
```
Comment (blindpirate) [inline subprojects/build-option/src/main/java/org/gradle/internal/buildoption/BooleanBuildOption.java]: Nitpicking: an extra line.

Comment (bmuschko) [reply subprojects/build-option/src/main/java/org/gradle/internal/buildoption/BooleanBuildOption.java]: done

**Pull request merged**